### PR TITLE
Named regexes, cask integration, more tests 

### PR DESCRIPTION
**Pull request by MC-Escherichia**
I did all this work to figure out the simple syntax issue for mixin highlighting, but I think it will aid others in future development.

### CODE REVIEW

**Comment by brianc** [comment]
@tjefferson08 has been maintaining this repo lately - I don't use emacs anymore.  I'm happy to merge this if you think it's good, but I'm flying pretty blind to be honest.

**Comment by tjefferson08** [comment]
Oh! @brinac I reached out to @MC-Escherichia via email but neglected to follow up on the GH issue here.

I think the PR looks great, I just wanted to pull down the changes and check them locally before merging. It's on my list for tomorrow! :) Thanks for the patience.

**Comment by brianc** [comment]
No problemo!  Keep on keepin' on!  Feel free to ping me if you need anything. :)

File: Makefile [hunk 1]
```diff
@@ -1,4 +1,4 @@
 test:
-	emacs -batch -L . -l ./tests/highlight.el -l ./tests/indentation.el -f ert-run-tests-batch-and-exit
+	cask exec ert-runner -l jade-mode.el
```
Comment (tjefferson08) [inline Makefile]: On my local and in [TCI](https://travis-ci.org/brianc/jade-mode/builds/73241865) `make test` gives me ```make: `test' is up to date.```

Applying the fix from [this SO question](http://stackoverflow.com/questions/3931741/why-does-make-think-the-target-is-up-to-date) solved it for me locally, could you make that change in this PR please?

File: example.jade [hunk 2]
```diff
@@ -28,6 +28,7 @@ html(lang="en", class = ['classOne', 'classTwo'].join(','))
           span this one shouldn't higlight strings or... .other #things if else
           | this one shouldn't highlight strings, and the same goes for .keywords #ok ?
         div#paren.content.example(style = 'float: left;') Content .here #should be plain if for
+        +help("if this is .bolded #that's bad, we want a string","a second arg or else");
```
Comment (tjefferson08) [inline example.jade]: This looks good on my local, but if you change `+help` to `+helpMe`, the uppercase "M" has no syntax highlighting. I think the regexp that is matching and highlighting the mixin lines is `jade-tag-re` (which intentionally doesn't highlight uppercase, although that could be its own discussion).

It looks like adding your new `jade-mixin-re` somewhere in the `font-lock-keywords` list might resolve that issue.

**Comment by tjefferson08** [comment]
In the interest of productivity, I'm going to merge! I'll take care of the issues I pointed out in the review in a follow-up PR. Thanks a million for the contribution, @MC-Escherichia! Keep 'em coming

**Pull request merged**